### PR TITLE
Modify Approver

### PR DESCRIPTION
**Pull request by madhavmansuriya40**

### CODE REVIEW

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    this.equals = this.checkDifference(this.intialSelectedApprovers, this.selectedApprovers);
+  }
+
+  checkDifference(intialSelectedApprovers, selectedApprovers) {
+    return isEqual(intialSelectedApprovers, selectedApprovers);
+  }
+
+  ngOnInit() {
+
+    // this.filteredOptions$ = fromEvent(this.searchBarRef.nativeElement, 'keyup').pipe(
```
Comment (Julias0) [inline src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:114]: dei no dead code

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      switchMap(() => {
+        return this.orgUserService.getAllCompanyEouc();
+      }),
+      map(eouc => {
+        return this.orgUserService.excludeByStatus(eouc, 'DISABLED');
+      }),
+      map(eouc => {
+        return eouc.filter((approver) => {
+          if (this.approverList.indexOf(approver.ou.id) > -1) {
+            approver['checked'] = true;
```
Comment (Julias0) [inline src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:145]: use . notation

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            approver['checked'] = false;
+          }
+          return approver;
+        });
+      }),
+      map(eouc => {
+        eouc = eouc.filter(approver => !this.selectedApprovers.includes(approver));
+        return this.selectedApprovers.concat(eouc);
+      }),
+      tap(() => {
```
Comment (Julias0) [inline src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:157]: You should be doing this in finalize and not in tap

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts (hunk 2 above)
Comment (madhavmansuriya40) [reply src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:145]: sir ye property temp add kiaye he, that is not present in modal so `.` notation nai chelga

**Review by Julias0 — APPROVED** [review]

**Pull request closed without merging**